docs/migrations/index.md

```diff
@@ -0,0 +1,15 @@
+(migrations-page)=
+
+# Migration notes
+
+The following pages contain migration notes for each stable release of Nextflow:
```

Comment:
```suggestion
This section covers important information for migrating between stable Nextflow releases. It includes descriptions of new features, enhancements, breaking changes, depreciations, and other miscellaneous updates that may require action during a migration.

For a preview of upcoming changes, see the {ref}`preview <preview-page>` page. This preview is a working document that tracks the most recent modifications planned for the next stable release. As development progresses, the content may change.

Migration notes for stable releases:
```
Some musings, feel free to adapt or ignore.

Reply:
side question -- is depreciation vs deprecation just a British/American English thing?